fetcher.go

```diff
@@ -5,13 +5,15 @@ package auth0
 import (
 	"bytes"
 	"encoding/json"
+	"github.com/dgrijalva/jwt-go"
 	"io/ioutil"
 	"net/http"
 )
 
 // TokenFetcher implementers can fetch an auth0 token.
 type TokenFetcher interface {
 	Token(audience string) (string, error)
+	CachedToken(audience string) (string, error)
```

Comment:
I'd remove CachedToken() from the interface. See comment below about renaming CachedToken().